main.go

```diff
@@ -95,14 +113,116 @@ func enableCors(w *http.ResponseWriter) {
 
 func handleRequest() {
 
+	serverAddr, ok := os.LookupEnv("JOBS_SERVER_ADDR")
+	if !ok {
+		serverAddr = ":8081"
+	}
+
 	myRouter := mux.NewRouter().StrictSlash(true)
 
 	myRouter.HandleFunc("/", homePage)
 	myRouter.HandleFunc("/jobs", showAllJobs).Methods("GET")
 	myRouter.HandleFunc("/jobs", testPostJobs).Methods("POST")
-	log.Fatal(http.ListenAndServe(":8081", myRouter))
+
+	myRouter.HandleFunc("/auth/redirect", func(w http.ResponseWriter, r *http.Request) {
+
+		url := authConfig.AuthCodeURL(oauthState)
+
+		http.Redirect(w, r, url, http.StatusFound)
+
+	})
+
+	myRouter.HandleFunc("/auth/callback", func(w http.ResponseWriter, r *http.Request) {
+		code := r.FormValue("code")
+		state := r.FormValue("state")
+
+		if state != oauthState {
+			http.Error(w, "invalid oauth state", http.StatusInternalServerError)
+			return
+		}
+
+		token, err := authConfig.Exchange(oauth2.NoContext, code)
+		if err != nil {
+			fmt.Fprintf(w, "Error: %s", err.Error())
+			return
+		}
+
+		// teamID := token.Extra("team_id")
+		// user := token.Extra("user")
+		// fmt.Fprintf(w, "%s %v", teamID, user)
+
+		basicInfo, err := getBasicInfo(string(token.AccessToken))
+		if err != nil {
+			fmt.Fprintf(w, "Error: %s", err.Error())
+			return
+		}
+
+		err = json.NewEncoder(w).Encode(basicInfo)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+		}
+
+	})
+
+	apiRouter := myRouter.PathPrefix("/api/v1").Subrouter()
+	apiRouter.HandleFunc("/test", func(w http.ResponseWriter, r *http.Request) {
+
+		// Override the default header middleware
+		w.Header().Set("Content-Type", "application/json")
+
+		err := json.NewEncoder(w).Encode(true)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+		}
+
+	}).Methods("GET")
+
+	// Route to test CORS headers
+	apiRouter.Handle("/cors", http.NotFoundHandler()).Methods("GET")
+
+	apiRouter.Use(handlers.CORS())
+	apiRouter.Use(defaultContentType)
+
+	loggedRouter := handlers.LoggingHandler(os.Stdout, myRouter)
+
+	log.Print("Setup server on: ", serverAddr)
+	log.Fatal(http.ListenAndServe(serverAddr, loggedRouter))
 }
 
 func main() {
 	handleRequest()
 }
+
+func defaultContentType(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Add("Content-Type", jsonapi.MediaType)
+		next.ServeHTTP(w, r)
+	})
+}
+
+func getBasicInfo(accessToken string) (d map[string]interface{}, err error) {
+
+	// NOTE: DO NOT USE https://slack.com/api/auth.test as it always
+	// responds with "invalid_auth".  I'm guessing it's a legacy endpoint
```

Comment:
ah ok, so is it working with slack?

Reply:
Yes, I put that as a warning for myself (or another dev).  That endpoint returns an "invalid_auth" even with a valid token.  I sent a note to Slack but haven't heard anything.  There were some comments online about them switching token types so I'm assume it's for the legacy token but couldn't find any confirmation.  I confirm the token is working in `getBasicInfo` function.